scripts/backup/backup_tool.py

```diff
@@ -0,0 +1,296 @@
+    return datetime.now().isoformat()
+
+
+class Metadata:
+    @staticmethod
+    def sha256(file_path):
+        hash_sha = hashlib.sha256()
+        with open(file_path, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_sha.update(chunk)
+        return hash_sha.hexdigest()
+
+    @staticmethod
+    def update_metadata_file(snapshot_path, snapshot_metadata):
+        metadata_file = os.path.join(snapshot_path, 'snapshots', 'metadata.json')
+        if os.path.isfile(metadata_file):
+            logging.info(f"Updating existing metadata file '{metadata_file}'")
+            with open(metadata_file, 'r') as f:
+                metadata = json.load(f)
+        else:
+            logging.info(f"Creating new metadata file '{metadata_file}'")
+            metadata = []
+
+        formatted_metadata = {
+            "name": snapshot_metadata["name"],
+            "created_at": snapshot_metadata["created_at"],
+            "compress": snapshot_metadata["compress"],
+            "data": snapshot_metadata["data"]
+        }
+
+        metadata.append(formatted_metadata)
+
+        with open(metadata_file, 'w') as f:
+            json.dump(metadata, f, indent=4)
+
+    @staticmethod
+    def update_metadata_after_removal(snapshots_dir, removed_snapshots):
+        metadata_file = os.path.join(snapshots_dir, 'metadata.json')
+        if not os.path.isfile(metadata_file):
+            return
+
+        logging.info(f"Updating metadata file '{metadata_file}' after snapshot removal")
+        with open(metadata_file, 'r') as f:
+            metadata = json.load(f)
+
+        updated_metadata = [entry for entry in metadata if entry["name"] not in removed_snapshots]
+
+        with open(metadata_file, 'w') as f:
+            json.dump(updated_metadata, f, indent=4)
+
+    @staticmethod
+    def create_snapshot_json(data_dir, snapshot_subdir):
+        files = []
+        for root, _, filenames in os.walk(data_dir):
+            for filename in filenames:
+                file_path = os.path.join(root, filename)
+                rel_path = os.path.relpath(file_path, data_dir)
+                snapshot_rel_path = os.path.join(snapshot_subdir, rel_path).replace('\\', '/')
+                file_info = {
+                    "name": filename,
+                    "path": snapshot_rel_path,
+                    "sha": Metadata.sha256(file_path)
+                }
+                files.append(file_info)
+
+        return {"data": files}
+
+    @staticmethod
+    def create_compressed_snapshot_json(compressed_file, rel_path):
+        file_info = {
+            "name": os.path.basename(compressed_file),
+            "path": rel_path,
+            "sha": Metadata.sha256(compressed_file)
+        }
+
+        return {"data": file_info}
+
+
+class DaemonManager:
+    @staticmethod
+    def is_daemon_running(daemon_path):
+        process_name = os.path.basename(daemon_path)
+        if os.name == 'nt':
```

Comment:
I don't see any point supporting windows here

Reply:
`nt` is windows otherwise it is unix base and linux